### PR TITLE
HIPP-654_add_an_api_success

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: app/views/AddAnApiSuccessView.scala.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+@applicationLinkUrl(application: Application) = @{
+    routes.ApplicationDetailsController.onPageLoad(application.id).url
+}
+
+@layout(pageTitle = titleNoForm(messages("addAnApiSuccess.title"))) {
+
+    @govukPanel(Panel(
+        title = Text(messages("addAnApiSuccess.title")),
+        content = HtmlContent(messages("addAnApiSuccess.summary", api.title))
+    ))
```
Comment (PaulCDurham) [inline app/views/AddAnApiSuccessView.scala.html:44]: Could you try adding classes = "govuk-!-margin-bottom-9" as well as title and content? I think that'll work. Will space more like the prototype.

File: app/views/AddAnApiSuccessView.scala.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    @(application.name)
+    </a>
+}
+
+
+@applicationLinkUrl(application: Application) = @{
+    routes.ApplicationDetailsController.onPageLoad(application.id).url
+}
+
+@layout(pageTitle = titleNoForm(messages("addAnApiSuccess.title"))) {
```
Comment (PaulCDurham) [inline app/views/AddAnApiSuccessView.scala.html:39]: Could you add showBackLink = false after pageTitle? We don't want a back link on this page. Also pass in the user so we can see who's logged in.

**Review by PaulCDurham — CHANGES_REQUESTED** [review]

File: test/controllers/AddAnApiSuccessControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        bind[Navigator].toInstance(new FakeNavigator(nextPage)),
+        bind[Clock].toInstance(clock)
+      )
+      .build()
+
+    val controller = application.injector.instanceOf[AddAnApiSuccessController]
+    Fixture(application, apiHubService, addAnApiSessionRepository, controller)
+  }
+
+  private def buildUserAnswers(): UserAnswers = {
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply test/controllers/AddAnApiSuccessControllerSpec.scala]: Ok

File: test/controllers/AddAnApiSuccessControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+  val app = mock[Application]
+  when(app.id).thenReturn("app_id")
+  when(app.name).thenReturn("app_name")
+
+  private val nextPage = routes.IndexController.onPageLoad
+  private val clock = Clock.fixed(Instant.now(), ZoneId.systemDefault())
+
+  "GET" - {
+    "must return OK and the correct view when the API detail exists for an unauthenticated user" in {
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply test/controllers/AddAnApiSuccessControllerSpec.scala]: Ok

File: test/controllers/AddAnApiSuccessControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+      running(fixture.application) {
+        val view = fixture.application.injector.instanceOf[AddAnApiSuccessView]
+
+        when(fixture.apiHubService.getApiDetail(ArgumentMatchers.eq(apiDetail.id))(any()))
+          .thenReturn(Future.successful(Some(apiDetail)))
+
+        when(fixture.apiHubService.getApplication(ArgumentMatchers.eq(app.id), any())(any()))
+          .thenReturn(Future.successful(Some(app)))
+
+        Console.println(s"URL: ${routes.AddAnApiSuccessController.onPageLoad(app.id, apiDetail.id).url}");
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply test/controllers/AddAnApiSuccessControllerSpec.scala]: Ok

File: test/controllers/AddAnApiSuccessControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+import java.time.{Clock, Instant, ZoneId}
+import scala.concurrent.Future
+
+class AddAnApiSuccessControllerSpec extends SpecBase with MockitoSugar with HtmlValidation with ApiDetailGenerators {
+
+  val apiDetail = mock[ApiDetail]
+  when(apiDetail.title).thenReturn("API title")
+  when(apiDetail.id).thenReturn("api_id")
+
+  val app = mock[Application]
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply test/controllers/AddAnApiSuccessControllerSpec.scala]: Ok

File: app/views/AddAnApiSuccessView.scala.html (hunk 1 above)
Comment (llmikeyj) [reply app/views/AddAnApiSuccessView.scala.html:46]: Ok

File: app/views/AddAnApiSuccessView.scala.html (hunk 2 above)
Comment (llmikeyj) [reply app/views/AddAnApiSuccessView.scala.html]: Ok

File: test/controllers/AddAnApiSuccessControllerSpec.scala [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+import services.ApiHubService
+import utils.HtmlValidation
+import views.html.{AddAnApiSuccessView, ErrorTemplate}
+
+import java.time.{Clock, Instant, ZoneId}
+import scala.concurrent.Future
+
+class AddAnApiSuccessControllerSpec extends SpecBase with MockitoSugar with HtmlValidation with ApiDetailGenerators {
+
+  val apiDetail = mock[ApiDetail]
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply test/controllers/AddAnApiSuccessControllerSpec.scala]: Ok

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**